src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -541,7 +541,14 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         ASR::array_index_t ai;
         ai.loc = x.base.base.loc;
         ai.m_left = nullptr;
-        ai.m_right = index;
+        // ASR uses 1-based indexing
+        ASR::ttype_t *a_type = ASRUtils::TYPE(ASR::make_Integer_t(al, x.base.base.loc,
+            4, nullptr, 0));
+        ASR::expr_t *constant_one = ASR::down_cast<ASR::expr_t>(ASR::make_ConstantInteger_t(
+                                            al, x.base.base.loc, 1, a_type));
+        make_BinOp_helper(index, constant_one, ASR::binopType::Add,
+                            x.base.base.loc, false);
+        ai.m_right = ASRUtils::EXPR(tmp);
```

Comment:
I think this change is not correct --- when we index, we represent it exactly as it is. 

Then for each array we store the lower and upper bound.